dev/Kozani/KozaniRemoteManager/KozaniDvcServer.cpp

```diff
@@ -0,0 +1,734 @@
+                    "Failed to register process lifetime tracking thread after the app is activated. Process Id: %u", processId) };
+                if (SUCCEEDED(hrRegisterWait))
+                {
+                    appActivatedAndTracked = true;
+                }
+            }
+            else
+            {
+                // The process Id is already in the m_processIdMap, which means the app activation does not create a new process.
+                // Apps can be single instance, which will use the same process to respond to multiple activations. 
+                // Remove the activity entry from the m_activityMap as we will not track duplicated activities backed by the same process.
+                g_connectionManager.m_activityMap.erase(iter);
+            }
+        }
+
+        SendActivateAppResult(pdu.activity_id(), S_OK, processId, isNewInstance);
+        return S_OK;
+    }
+    CATCH_RETURN()
+
+    HRESULT KozaniDvcServer::ProcessAppTerminationNotice(_In_ Dvc::ProtocolDataUnit& pdu) noexcept try
+    {
+        auto lock{ g_connectionManager.m_activityMapLock.lock_exclusive() };
+        RETURN_HR_IF(HRESULT_FROM_WIN32(ERROR_INVALID_STATE), g_connectionManager.m_closing);
+
+        auto iter{ g_connectionManager.m_activityMap.find(pdu.activity_id()) };
+        if (iter == g_connectionManager.m_activityMap.end())
+        {
+            RETURN_HR_MSG(HRESULT_FROM_WIN32(ERROR_NOT_FOUND), "The activity Id %I64u does not exist in the activity map", pdu.activity_id());
+        }
+
+        auto& appInfo{ iter->second };
+        appInfo.activationStatus = AppActivationStatus::TerminationRequestedFromClient;
+
+        if (appInfo.pid == 0)
+        {
+            // The app activation has not finished yet. It will be terminated immediately upon activation since it is 
+            // marked with AppActivationStatus::TerminationRequestedFromClient. 
+            return S_OK;
+        }
+
+        FAIL_FAST_IF_MSG(!appInfo.processHandle, "Process handle should be valid when pid is not zero");
+
+        HRESULT hrTerminateProcess{ LOG_IF_WIN32_BOOL_FALSE_MSG(TerminateProcess(appInfo.processHandle.get(), 0), "Failed to terminate process %u", appInfo.pid) };
+
+        if (!appInfo.processLifetimeTrackerHandle)
+        {
+            // No process lifetime tracker to handle at ProcessTerminationCallback - we should clean up the maps here.
+            g_connectionManager.m_processIdMap.erase(appInfo.pid);
+            g_connectionManager.m_activityMap.erase(iter);
+        }
+
+        RETURN_HR(hrTerminateProcess);
+    }
+    CATCH_RETURN()
+
+    bool KozaniDvcServer::IsActivationKindSupported(Dvc::ActivationKind kind)
+    {
+        switch (kind)
+        {
+            case Dvc::ActivationKind::Launch:
+            case Dvc::ActivationKind::File:
+            case Dvc::ActivationKind::Protocol:
+                return true;
+        }
+        return false;
```

Comment:
Probably better to avoid switch here e.g.

```
bool IsSupported(kind)
{
     return (kind == Dvc::ActivationKind::Launch) || (kind == Dvc::ActivationKind::File) || (kind == Dvc::ActivationKind::Protocol);
}
```

Switch isn't always the most efficient, and can get worse for larger cases. Plus just general readability the switch is more verbose for no real gain

Reply:
Switch is more efficient than long list of conditions and easier to read if there are many conditions. Compiler optimizes switch statement using jump table so it can be O(1) complexity rather than O(n) for if statements with n conditions (cases).